auth/overview/api_key.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package overview contains Google Cloud authentication overview snippets.
+// https://cloud.google.com/docs/authentication/
+package overview
+
+import (
+	"context"
+	"log"
+
+	"google.golang.org/api/option"
+	"google.golang.org/api/pubsub/v1"
+)
+
+// [START auth_overview_api_key]
+
+// apiKey shows how to use API key to authenticate.
+func apiKey() {
+	service, err := pubsub.NewService(context.Background(),
+		option.WithAPIKey("api-key-string"))
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Please return errors rather than calling `log.Fatal`.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#return-errors